src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseParticipationResource.java

```diff
@@ -82,6 +86,29 @@ public ProgrammingExerciseParticipationResource(ProgrammingExerciseParticipation
         }
     }
 
+    /**
+     * GET /programming-exercise-participation/:id/latest-pending-submission : get the latest pending submission for the participation.
+     * A pending submission is one that does not have a result yet and is not older than RESULT_WAIT_LIMIT_SECONDS.
+     *
+     * @param participationId the id of the participation get the latest submission for
+     * @return the ResponseEntity with status 200 (OK)
+     */
+    @GetMapping("/programming-exercise-participation/{participationId}/latest-pending-submission")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ProgrammingSubmission> getLatestPendingSubmission(@PathVariable Long participationId) {
+        ProgrammingSubmission submission;
+        try {
+            submission = submissionService.getLatestPendingSubmission(participationId);
```

Comment:
what happens if this is null? This might be the "normal" case, so I would like to avoid that we have a lot of cases with 403 not found. Can we return "null" in this case?

Reply:
We might then need to update the JavaDoc above the method

Reply:
It will return null if there is no pending submission.

Reply:
I updated the doc.